gateway/join.go

```diff
@@ -81,9 +81,8 @@ func (g *gateway) handleJoin(ctx context.Context, payload []byte) error {
 	txPkt.payload = cPayload
 
 	// send on rx2 window - opens 6 seconds after join request.
-	if !utils.SelectContextOrWait(ctx, time.Second*joinRx2WindowSec) {
-		return nil
-	}
+	waitDuration := (joinRx2WindowSec * time.Second) - (time.Since(t))
+	accurateSleep(ctx, waitDuration)
```

Comment:
[nit] have a comment that we will move this to the generic downlink function when that is ready